cmd/kubeadm/app/cmd/init.go

```diff
@@ -383,7 +383,7 @@ func (i *Init) Run(out io.Writer) error {
 	}
 
 	// PHASE 4: Mark the master with the right label/taint
-	if err := markmasterphase.MarkMaster(client, i.cfg.NodeName); err != nil {
+	if err := markmasterphase.MarkMaster(client, i.cfg.NodeName, !i.cfg.NoTaintMaster); err != nil {
```

Comment:
Should add a `BoolVar` flag `no-taint-master` to `AddInitConfigFlags`.
If not, `cfg.NoTaintMaster` will always be false and cannot be override by parsing arguments.

Reply:
`cfg.NoTaintMaster` can be set in the config file. For testing I have `/etc/kubeadm/kubeadm.yaml` containing:
```yaml
apiVersion: kubeadm.k8s.io/v1alpha1
kind: MasterConfiguration
kubernetesVersion: v1.9.1
noTaintMaster: true
```
and am running `kubeadm init --config /etc/kubeadm/kubeadm.yaml`.

My understanding was that the config file mechanism was the future and the CLI variants were deprecated, perhaps I understood wrongly? I'll add the lines to `AddInitConfigFlags`, it looks like it will be pretty straightforward.

Reply:
> My understanding was that the config file mechanism was the future and the CLI variants were deprecated

I agree. But for now, we still use arguments. @luxas WDYT?